src/OrleansRuntime/Catalog/IGrainActivationContext.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using Orleans.Core;
+
+namespace Orleans.Runtime
+{
+    /// <summary>
+    /// The context associated with the current grain activation.
+    /// </summary>
+    public interface IGrainActivationContext
+    {
+        /// <summary>Gets the .NET type of the grain activation instance.</summary>
+        Type GrainType { get; }
+
+        /// <summary>Gets the identity of the grain activation.</summary>
+        IGrainIdentity GrainIdentity { get; }
+
+        /// <summary>Gets the <see cref="IServiceProvider"/> that provides access to the grain's service container.</summary>
+        IServiceProvider GrainServices { get; }
```

Comment:
I just not fond of loosing the "Provider" suffix, but as we talked ApplicationServices maybe a better name?